pkg/controller/direct/gkehub/featuremembership_controller.go

```diff
@@ -32,7 +33,11 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
 )
 
-const ctrlName = "gkehubfeaturemembership-controller"
+const (
+	ctrlName    = "gkehubfeaturemembership-controller"
+	timeout     = 5 * time.Minute
```

Comment:
Just curious if 5 min is enough? I remember gkehubfeaturemembership is a long running resources(takes more than 10min). Shall we have special handling for long running resources?

Reply:
/cc @shumiao 
The timeout is for a GKEHubfeature Update call,
What is the recommended timeout period for such operation?

Reply:
We don't want to do this in a direct controller. Currently you can just get it pass through like the cbwp [controller](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/direct/cloudbuild/workerpool_controller.go#L191C2-L191C9). We can improve this later

Reply:
The cbwp controller uses `https://github.com/googleapis/google-cloud-go/tree/main/cloudbuild` to create GCP client. However, the corresponding `gkehub` one misses gkehub feature fields in the repo https://github.com/googleapis/google-cloud-go/tree/main/gkehub. 

By reading the repo description, https://screenshot.googleplex.com/57dR6itytjCs6Ls seems like the repo is still under development

The gkehub controller uses `https://github.com/googleapis/google-api-go-client/tree/main/gkehub/v1beta`, which doesn't have `Wait` built-in

Reply:
Gemma is right that the long running operation's TTL is 20 minutes, meaning in the worst case, it will take that long be marked done, with either a success or a failure code. The client doesn't have `wait` built-in. This is the client side `wait` logic internal google3 controllers share: https://source.corp.google.com/piper///depot/google3/cloud/kubernetes/hub/common/hubclient/tools.go;l=38-50;rcl=657751230. If I understand correctly, they don't set a timeout, but wait for it to complete or until the context times out.

Reply:
@yuwenma Friendly ping, how about I go ahead implement the Wait similar to what shumiao shared?

Reply:
Yes, I think you just poll the operation every 2 seconds (for example).  I wouldn't bother to do a backoff, because we're going to make this much better soon (return from the reconciler etc)

If the operation finished with an error, return an error.  If not, return with no error.  controller-runtime will run the retry for you.

Reply:
Thanks for providing additional context here. @justinsb  Now the controller polls the operation every 5s and times out after 20 min, which mimics what the Hub client does today. Could you review this PR?